src/Authoring/WinRT.SourceGenerator/DiagnosticUtils.cs

```diff
@@ -339,8 +339,8 @@ private bool IsInvalidNamespace(INamespaceSymbol @namespace, string assemblyName
             {
                 topLevel = topLevel.ContainingNamespace;
```

Comment:
Shouldn't we also be checking if AssemblyName != partial top level rather than the most top level only?